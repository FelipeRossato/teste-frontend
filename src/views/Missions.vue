<template>
  <div class="container">
    <div v-if="loading">
      <Shimmer :mode="mode" />
    </div>
    <div v-else>
      <Header 
        @toggle="toggleTheme()" 
        :mode="mode" 
      />
      <MissionsFilter 
        :mode="mode"
        :missions="missions"
      />    
    </div>
  </div>
  <router-view />
</template>

<script>
import Shimmer from '../components/Shimmer';
import Header from '../components/Header';
import MissionsFilter from '../components/MissionsFilter';
import axios from 'axios';

export default {
  name: 'Missions',
  components: {
    Shimmer,
    Header,
    MissionsFilter,
  },

  data() {
    return {
      mode: 'light',
      missions: [],
      loading: true
    }
  },

  async created() {
    // Fetch no endpoint utilizando axios
    const response = await axios.get('https://us-central1-teste-frontend-c2dcc.cloudfunctions.net/missions')
    this.missions = response.data
    this.loading = false 
  },

  // DARK/LIGHT TOGGLE MODE
  methods: {
    toggleTheme() {
      localStorage.setItem('mode', (localStorage.getItem('mode') || 'light') === 'light' ? 'dark' : 'light');
      localStorage.getItem('mode') === 'dark' ? 
      [this.mode = 'dark', document.body.classList.add('dark')] : 
      [this.mode = 'light', document.body.classList.remove('dark'), document.body.classList.add('light')]
    } 
  },
  
  mounted() {
    if (localStorage.mode === 'dark') {
      this.mode = 'dark' 
      document.body.classList.add('dark')
    } 
  }
}
</script>

<style>
  @import url('https://fonts.googleapis.com/css2?family=Merriweather+Sans:ital,wght@0,400;0,700;1,500&display=swap');

  :root {
    --body-color: #f7f7f7;
    --DM-body-color: #333333;

    --main-color: #00CEB2;

    --title-color: #078473;

    --text-color: #5d5d5d;
    --DM-text-color: #B5B5B5;

    --background-color: rgba(219, 218, 218, 0.6);
    --DM-background-color: rgba(67, 67, 67, 0.6);

    --lightBackground-color: rgba(219, 218, 218, 0.4);
    --DM-lightBackground-color: rgba(67, 67, 67, 0.4);

    --box-shadow: 0px 5px 5px 1px rgba(169, 169, 169, 0.7);
    --DM-box-shadow: 0px 5px 5px 1px rgba(0, 0, 0, 0.7);

    --button-color: #078473;

    --tag-emAndamento-color: #7D5F00;
    --tag-emAndamento-bgColor: rgba(255, 219, 104, 0.5);

    --DM-tag-emAndamento-color: #5d4800;
    --DM-tag-emAndamento-bgColor: rgba(255, 219, 104);

    --tag-naoIniciada-color: #004EB2;
    --tag-naoIniciada-bgColor: rgba(138, 186, 248, 0.5);

    --DM-tag-naoIniciada-color: #002c66;
    --DM-tag-naoIniciada-bgColor: rgba(138, 186, 248);

    --tag-completa-color: #006A0B;
    --tag-completa-bgColor: rgba(121, 251, 135, 0.5);

    --DM-tag-completa-color: #004d08;
    --DM-tag-completa-bgColor: rgba(121, 251, 135);

    --tag-bloqueada-color: #505050;
    --tag-bloqueada-bgColor: rgba(158, 158, 158, 0.5);

    --DM-tag-bloqueada-color: #242424;
    --DM-tag-bloqueada-bgColor: rgba(158, 158, 158);

    --red-color: #af0101;
    --DM-red-color: rgb(255, 167, 167);

    --white-color: white;
    --black-color: black;

    --bar-color: #8AE993;
    --DM-bar-color: #55d862;

    --shimmer-color: rgb(219, 218, 218);
    --DM-shimmer-color: rgb(102, 102, 102);
  }

  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }

  body {
    font-family: 'Merriweather Sans', sans-serif;
    max-width: 1050px;
    margin: 0 auto;
    background-color: var(--body-color);
    transition: background-color 0.3s ease-in-out;
  }

  body.dark {
    background-color: var(--DM-body-color);
  }

  .container {
    display: flex;
    flex-direction: column;
  }

</style>