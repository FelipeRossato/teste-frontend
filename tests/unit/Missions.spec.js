import { mount } from '@vue/test-utils'
import Missions from '@/views/Missions.vue'
import missionsData from '@/data/missions.js'
import axios from 'axios'

// Mock da lista de missions do endpoint. Length: 6 
const mockMissions = missionsData;

jest.mock('axios', () => ({
  get: () => Promise.resolve({data: mockMissions}) 
}))

describe('Missions', () => {
  let wrapper;

  beforeEach(() => {
    // Dependência 'jest-localstorage-mock' utilizada para fazer um mock de test do localStorage
    localStorage.clear();

    wrapper = mount(Missions, {
      mocks: {
        axios
      }
    })

  })

  test('Renderização inicial e snapshot', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  test('Verifica se o valor do localStorage está limpo', () => {
    expect(localStorage.getItem('mode')).toBeNull();
  })
  
  test('Verifica se a função que troca o light/dark mode está funcionando', () => {
    wrapper.vm.toggleTheme()
    expect(wrapper.vm.mode).toBe('dark')
    expect(localStorage.getItem('mode')).toBe('dark');
    wrapper.vm.toggleTheme()
    expect(wrapper.vm.mode).toBe('light')
    expect(localStorage.getItem('mode')).toBe('light');
  })

  test('Verifica se as missions estão sendo renderizadas no component', () => {
    expect(wrapper.vm.missions.length).toBe(6)
    const missionDiv = wrapper.findAll('.mission-template')
    expect(missionDiv.length).toBe(6)
  })

})







