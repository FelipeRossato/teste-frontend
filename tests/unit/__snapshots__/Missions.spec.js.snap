// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Renderização inicial e snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__props": Array [],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      
      <div
        class="container"
      >
        <div>
          <div
            class="header"
          >
            <div
              class="logo-wrapper"
            >
              <a
                href="https://skore.io/pt"
              >
                <img
                  alt="Skore Logo"
                />
              </a>
            </div>
            <nav
              class="navbar"
            >
              <a
                href="/"
              >
                <img
                  alt="Home Icon"
                />
              </a>
              <a
                href="#"
              >
                <img
                  alt="User Icon"
                />
              </a>
              <div
                id="toggleBtn"
              >
                <img
                  alt="DarkMode Icon"
                  src=""
                />
              </div>
            </nav>
          </div>
          
          <h1
            class="light"
            id="missions"
          >
             MISSÕES 
          </h1>
          <div
            class="filter-wrapper light"
          >
            <input
              id="todas"
              name="Filter"
              type="radio"
              value="ALL"
            />
            <p>
              <label
                for="todas"
              >
                Todas
              </label>
            </p>
            <input
              id="em-andamento"
              name="Filter"
              type="radio"
              value="IN_PROGRESS"
            />
            <p>
              <label
                for="em-andamento"
              >
                Em Andamento
              </label>
            </p>
            <input
              id="completa"
              name="Filter"
              type="radio"
              value="COMPLETED"
            />
            <p>
              <label
                for="completa"
              >
                Completa
              </label>
            </p>
            <input
              id="nao-iniciada"
              name="Filter"
              type="radio"
              value="NOT_STARTED"
            />
            <p>
              <label
                for="nao-iniciada"
              >
                Não Iniciada
              </label>
            </p>
            <input
              id="bloqueada"
              name="Filter"
              type="radio"
              value="BLOCKED"
            />
            <p>
              <label
                for="bloqueada"
              >
                Bloqueada
              </label>
            </p>
          </div>
          <div
            class="list-wrapper"
          >
            
            <div>
              <div
                class="mission-template light"
              >
                <div
                  class="no-overlay"
                />
                <div
                  class="tags"
                >
                  <span
                    class="status em-andamento light"
                  >
                    Em Andamento
                  </span>
                  <div
                    class="priority light"
                  >
                     Prioridade 
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="red-circle"
                      />
                    </div>
                    
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    
                  </div>
                </div>
                <div
                  class="image-wrapper"
                >
                  <img
                    alt=""
                    src="https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png"
                  />
                </div>
                <div
                  class="info-wrapper"
                >
                  <h1
                    class="light"
                  >
                    Desenvolvedor Backend
                  </h1>
                  <div
                    class="meter"
                  >
                    <span
                      class="bar light"
                      style="--width: 21%;"
                    />
                  </div>
                  <div
                    class="btn"
                  >
                    <router-link
                      to="[object Object]"
                    >
                       ACESSAR 
                    </router-link>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="mission-template light"
              >
                <div
                  class="no-overlay"
                />
                <div
                  class="tags"
                >
                  <span
                    class="status nao-iniciado light"
                  >
                    Não Iniciada
                  </span>
                  <div
                    class="priority light"
                  >
                     Prioridade 
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="red-circle"
                      />
                    </div>
                    
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    
                  </div>
                </div>
                <div
                  class="image-wrapper"
                >
                  <img
                    alt=""
                    src="https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png"
                  />
                </div>
                <div
                  class="info-wrapper"
                >
                  <h1
                    class="light"
                  >
                    Desenvolvedor Frontend
                  </h1>
                  <div
                    class="meter"
                  >
                    <span
                      class="bar light"
                      style="--width: undefined%;"
                    />
                  </div>
                  <div
                    class="btn"
                  >
                    <router-link
                      to="[object Object]"
                    >
                       ACESSAR 
                    </router-link>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="mission-template light"
              >
                <div
                  class="no-overlay"
                />
                <div
                  class="tags"
                >
                  <span
                    class="status em-andamento light"
                  >
                    Em Andamento
                  </span>
                  <div
                    class="priority light"
                  >
                     Prioridade 
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="red-circle"
                      />
                    </div>
                    
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    
                  </div>
                </div>
                <div
                  class="image-wrapper"
                >
                  <img
                    alt=""
                    src="https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png"
                  />
                </div>
                <div
                  class="info-wrapper"
                >
                  <h1
                    class="light"
                  >
                    Desenvolvedor iOS
                  </h1>
                  <div
                    class="meter"
                  >
                    <span
                      class="bar light"
                      style="--width: 63%;"
                    />
                  </div>
                  <div
                    class="btn"
                  >
                    <router-link
                      to="[object Object]"
                    >
                       ACESSAR 
                    </router-link>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="mission-template light"
              >
                <div
                  class="completed-overlay"
                />
                <div
                  class="tags"
                >
                  <span
                    class="status completa light"
                  >
                    Completa
                  </span>
                  <div
                    class="priority light"
                  >
                     Prioridade 
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="red-circle"
                      />
                    </div>
                    
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    
                  </div>
                </div>
                <div
                  class="image-wrapper"
                >
                  <img
                    alt=""
                    src=""
                  />
                </div>
                <div
                  class="info-wrapper"
                >
                  <h1
                    class="light"
                  >
                    Desenvolvedor Android
                  </h1>
                  <div
                    class="meter"
                  >
                    <span
                      class="bar light"
                      style="--width: 100%;"
                    />
                  </div>
                  <div
                    class="btn"
                  >
                    <router-link
                      to="[object Object]"
                    >
                       ACESSAR 
                    </router-link>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="mission-template light"
              >
                <div
                  class="no-overlay"
                />
                <div
                  class="tags"
                >
                  <span
                    class="status nao-iniciado light"
                  >
                    Não Iniciada
                  </span>
                  <div
                    class="priority light"
                  >
                     Prioridade 
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="red-circle"
                      />
                    </div>
                    
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    
                  </div>
                </div>
                <div
                  class="image-wrapper"
                >
                  <img
                    alt=""
                    src="https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png"
                  />
                </div>
                <div
                  class="info-wrapper"
                >
                  <h1
                    class="light"
                  >
                    Desenvolvedor Flutter
                  </h1>
                  <div
                    class="meter"
                  >
                    <span
                      class="bar light"
                      style="--width: undefined%;"
                    />
                  </div>
                  <div
                    class="btn"
                  >
                    <router-link
                      to="[object Object]"
                    >
                       ACESSAR 
                    </router-link>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="mission-template light"
              >
                <div
                  class="blocked-overlay"
                />
                <div
                  class="tags"
                >
                  <span
                    class="status bloqueada light"
                  >
                    Bloqueada
                  </span>
                  <div
                    class="priority light"
                  >
                     Prioridade 
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="red-circle"
                      />
                    </div>
                    
                    
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    <div
                      class="priority-circle"
                    >
                      <span
                        class="white-circle"
                      />
                    </div>
                    
                  </div>
                </div>
                <div
                  class="image-wrapper"
                >
                  <img
                    alt=""
                    src="https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png"
                  />
                </div>
                <div
                  class="info-wrapper"
                >
                  <h1
                    class="light"
                  >
                    Site Reliability Engineer
                  </h1>
                  <div
                    class="meter"
                  >
                    <span
                      class="bar light"
                      style="--width: undefined%;"
                    />
                  </div>
                  <div
                    class="btn"
                  >
                    <router-link
                      to="[object Object]"
                    >
                       ACESSAR 
                    </router-link>
                  </div>
                </div>
              </div>
            </div>
            
          </div>
          
        </div>
      </div>
      <router-view />
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0-rc.6",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "loading": false,
    "missions": Array [
      Object {
        "company_id": "114",
        "due_date": Object {
          "available_at": 1572127435329,
        },
        "enrollment": Object {
          "id": "114_15a9LAUXCm9FkAVexH4q_453995",
          "percentage": 21,
        },
        "id": "15a9LAUXCm9FkAVexH4q",
        "name": "Desenvolvedor Backend",
        "priority": 1,
        "status": "IN_PROGRESS",
        "thumb_url": "https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png",
      },
      Object {
        "company_id": "114",
        "due_date": Object {
          "available_at": 1577460755435,
        },
        "id": "OgYPcZXbYgHQCFCS3b8a",
        "name": "Desenvolvedor Frontend",
        "priority": 1,
        "status": "NOT_STARTED",
        "thumb_url": "https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png",
      },
      Object {
        "company_id": "114",
        "due_date": null,
        "enrollment": Object {
          "id": "114_0DynFIwidEoqpJxbpmQK_453995",
          "percentage": 63,
        },
        "id": "0DynFIwidEoqpJxbpmQK",
        "name": "Desenvolvedor iOS",
        "priority": 1,
        "status": "IN_PROGRESS",
        "thumb_url": "https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png",
      },
      Object {
        "company_id": "114",
        "due_date": null,
        "enrollment": Object {
          "id": "114_15a9LAUXCm9FkAVexH4q_453995",
          "percentage": 100,
        },
        "id": "15a9LAUXCm9FkAVexH4q",
        "name": "Desenvolvedor Android",
        "priority": 1,
        "status": "COMPLETED",
        "thumb_url": null,
      },
      Object {
        "company_id": "114",
        "due_date": null,
        "id": "Jx0TJYrLFhYOeIc4boBL",
        "name": "Desenvolvedor Flutter",
        "priority": 1,
        "status": "NOT_STARTED",
        "thumb_url": "https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png",
      },
      Object {
        "company_id": "114",
        "due_date": null,
        "id": "9LMvnEZFlnmwoIjCNqHL",
        "name": "Site Reliability Engineer",
        "priority": 1,
        "status": "BLOCKED",
        "thumb_url": "https://img.skore.io/image/upload/v1566918438/retool/c6mjszo.png",
      },
    ],
    "mode": "light",
    "toggleTheme": [Function],
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
